packages/@atjson/source-url/src/converter.ts

```diff
@@ -1,26 +1,23 @@
 import OffsetSource, { SocialURLs } from "@atjson/offset-annotations";
-import { URLAnnotation } from "./annotations";
 import URLSource from "./source";
 
-URLSource.defineConverterTo(OffsetSource, doc => {
-  doc
-    .where((annotation: URLAnnotation) => annotation.type === "url")
-    .update((annotation: URLAnnotation) => {
-      let { url, AnnotationClass } = SocialURLs.identify(annotation.attributes);
-      if (url && AnnotationClass) {
-        doc.replaceAnnotation(
-          annotation,
-          new AnnotationClass({
-            id: annotation.id,
-            start: annotation.start,
-            end: annotation.end,
-            attributes: {
-              url
-            }
-          })
-        );
-      }
-    });
+URLSource.defineConverterTo(OffsetSource, function urlToOffset(doc) {
+  for (let annotation of doc.where({ type: "-whatwg-url" }).annotations) {
```

Comment:
So, this will be harder to migrate to the new syntax that you were proposing for converters, but this is ok.

Maybe don't optimize this one, since it runs against a single string of text 🙃